File: mobile_app/lib/services/patient_service.dart
import '../core/api_client.dart';
import '../models/doctor.dart';
import '../models/entry.dart';
import '../models/patient_profile.dart';

/// GET /patients/{id}, GET /patients/{id}/symptom-checklist, and
/// POST+GET /patients/{id}/notes — see context/api-contracts.md. The notes
/// methods were added for the doctor role (2026-08-29) — they live here
/// rather than in DoctorService since api-contracts.md groups
/// `/patients/{id}/notes` under its "Patients" section (same URL prefix as
/// the other two methods already here), even though only doctor/admin
/// callers can actually reach them (RBAC enforced backend-side).
class PatientService {
  final _dio = ApiClient.instance.dio;

  Future<PatientProfile> getPatient(int patientId) {
    return ApiClient.instance.run(() async {
      final res = await _dio.get('/patients/$patientId');
      return PatientProfile.fromJson(res.data as Map<String, dynamic>);
    });
  }

  Future<List<SymptomOption>> getSymptomChecklist(int patientId) {
    return ApiClient.instance.run(() async {
      final res = await _dio.get('/patients/$patientId/symptom-checklist');
      return (res.data as List<dynamic>)
          .map((e) => SymptomOption.fromJson(e as Map<String, dynamic>))
          .toList();
    });
  }

  Future<List<DoctorNote>> getNotes(int patientId) {
    return ApiClient.instance.run(() async {
      final res = await _dio.get('/patients/$patientId/notes');
      return (res.data as List<dynamic>)
          .map((e) => DoctorNote.fromJson(e as Map<String, dynamic>))
          .toList();
    });
  }

  Future<DoctorNote> addNote(int patientId, String noteText) {
    return ApiClient.instance.run(() async {
      final res = await _dio.post('/patients/$patientId/notes', data: {'note_text': noteText});
      return DoctorNote.fromJson(res.data as Map<String, dynamic>);
    });
  }
}

File: web_app/src/services/patientService.js
import { apiClient } from '../core/apiClient'

// GET /patients/{id}, GET /patients/{id}/symptom-checklist,
// GET+POST /patients/{id}/notes — mirrors
// mobile_app/lib/services/patient_service.dart. The notes methods are
// doctor/admin-only backend-side (RBAC enforced server-side, not here) —
// kept in this file rather than doctorService.js since api-contracts.md
// groups /patients/{id}/notes under "Patients", same URL prefix as this
// file's other two methods, same reasoning as the Flutter file's own doc
// comment. Added 2026-08-29 (React doctor-role pass) — the doctor role was
// previously the reason these were left out.
export const patientService = {
  getPatient(patientId) {
    return apiClient.get(`/patients/${patientId}`)
  },

  getSymptomChecklist(patientId) {
    return apiClient.get(`/patients/${patientId}/symptom-checklist`)
  },

  getNotes(patientId) {
    return apiClient.get(`/patients/${patientId}/notes`)
  },

  addNote(patientId, noteText) {
    return apiClient.post(`/patients/${patientId}/notes`, { note_text: noteText })
  },
}
